src/Common/src/Interop/Unix/System.Native/Interop.GetGroupList.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal unsafe static uint[] GetGroupList(string userName, uint primaryGroupId)
+        {
+            Span<uint> groups = stackalloc uint[64];
+            do
+            {
+                int rv;
+                int ngroups = groups.Length;
+                fixed (uint* pGroups = groups)
+                {
+                    rv = Interop.Sys.GetGroupList(userName, primaryGroupId, pGroups, &ngroups);
+                }
+                if (rv >= 0)
+                {
+                    // success
+                    return groups.Slice(0, ngroups).ToArray();
+                }
+                else if (rv == -1 && ngroups > groups.Length)
+                {
+                    // increase buffer size
+                    groups = new uint[groups.Length * 2];
```

Comment:
Does ngroups end up meaning the number of groups in the current list?  If so, should this instead be `Math.Max(ngroups, groups.Length * 2)`?